tsdb/head.go

```diff
@@ -414,17 +418,24 @@ func (h *Head) processWALSamples(
 				continue
 			}
 			ms := refSeries[s.Ref]
-			if ms == nil {
-				ms = h.series.getByID(s.Ref)
-				if ms == nil {
+			if ms.ms == nil {
+				ms.ms = h.series.getByID(s.Ref)
+				if ms.ms == nil {
 					unknownRefs++
 					continue
 				}
+				if len(ms.ms.mmappedChunks) == 0 {
+					ms.minTime = -1
```

Comment:
I'm not sure, but I think I've heard TSDB has been designed to work making no assumptions on the timestamp value. @codesome can share more light on this.

```suggestion
					ms.minTime = math.MinInt64
```

Reply:
👍🏼